File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/UploadRepositoryImpl.kt
package com.ssafy.ssabree.core.repository

import com.ssafy.ssabree.core.datasource.remote.UploadService
import com.ssafy.ssabree.core.utils.RetrofitClient
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody.Companion.asRequestBody
import java.io.File

class UploadRepositoryImpl : UploadRepository {

    private val uploadService: UploadService = RetrofitClient.uploadInstance.create(UploadService::class.java)

    override suspend fun uploadImage(file: File): Result<String> {
        return runCatching {
            val requestBody = file.asRequestBody("image/jpeg".toMediaType())
            val multipart = MultipartBody.Part.createFormData(
                name = "file",
                filename = file.name,
                body = requestBody
            )
            uploadService.uploadImage(multipart).url
        }
    }
}

File: iOS/ssabree/ssabree/Core/Repository/UploadRepositoryImpl.swift
import Foundation

final class UploadRepositoryImpl: UploadRepository {
    private let uploadService: UploadService

    init(uploadService: UploadService) {
        self.uploadService = uploadService
    }

    func uploadImage(image: Data) async -> Result<String, Error> {
        do {
            let response = try await uploadService.uploadImage(image: image)
            return .success(response.url)
        } catch {
            return .failure(error)
        }
    }
}
